server/akka-http-server/src/main/scala/sttp/tapir/server/akkahttp/AkkaRequestBody.scala

```diff
@@ -31,7 +31,7 @@ private[akkahttp] class AkkaRequestBody(serverOptions: AkkaHttpServerOptions)(im
       case RawBodyType.StringBody(_)  => implicitly[FromEntityUnmarshaller[String]].apply(body).map(RawValue(_))
       case RawBodyType.ByteArrayBody  => implicitly[FromEntityUnmarshaller[Array[Byte]]].apply(body).map(RawValue(_))
       case RawBodyType.ByteBufferBody => implicitly[FromEntityUnmarshaller[ByteString]].apply(body).map(b => RawValue(b.asByteBuffer))
-      case RawBodyType.InputStreamBody =>
+      case RawBodyType.InputStreamBody | RawBodyType.ResourceBody =>
```

Comment:
First of all, this code concerns request body, which is not a case for the static content API. However, the `RawBodyType` type is also used in requests, so my guess here is that we need to write a handler here, even if it's not applicable?

Reply:
In fact ... maybe we should instead have a `InputStreamSupplier` raw body type, which would be more universal (and which we should have used instead of the `InputStream` one, but it's too late now ;) ). It should also be easier to support in the backends.

Reply:
At least we can rename `RawBodyType.ResourceBody`  to `InputStreamSupplier` for the new body type?
BTW I'm quite confused about how this works now.
```scala
     case RawBodyType.InputStreamBody =>
        implicitly[FromEntityUnmarshaller[Array[Byte]]].apply(body).map(b => RawValue(new ByteArrayInputStream(b)))
```
why is `FromEntityUnmarshaller[Array[Byte]]` applied to body here? Why can we assume that it's an `Array[Byte]`?

Reply:
Well the request body can always be represented as an `Array[Byte]` (assuming you've got enough RAM ;) ). I think in case of akka-streams it's not possible to get the body represented as an `InputStream`, so the only thing that we might do is read the body as an array of bytes, and fake an input stream on top of that